traffic_ops/traffic_ops_golang/select_stmt.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"net/url"
+
+	"github.com/apache/incubator-trafficcontrol/traffic_monitor_golang/common/log"
+)
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+*/
+
+func SelectStmt(v url.Values, selectStmt string, queryParamsToQueryCols map[string]string) (string, map[string]interface{}) {
+
+	var sqlQuery string
+	if len(v) > 0 {
+		sqlQuery = selectStmt + "\nWHERE " + appendCriteria(queryParamsToQueryCols, v)
```

Comment:
I think appendCriteria and queryValues should be a single function that returns a string and a map, since they each loop over the url.Values. appendCriteria even builds the map up still but it is just discarded. This also will create a broken sql statement if random, non-mapped queryParams are passed in

```
var sqlQuery string
var queryValues map[string]interface{}
if len(v) > 0 {
 	criteria, queryValues := parseCriteriaAndQueryValues(queryParamsToQueryCols, v) // I don't like that the scoping of queryValues looks a bit ambiguous here...
 	if len(queryValues) > 0 {
 		 sqlQuery = selectStmt + "\nWHERE " + criteria
 	} else {
 		 sqlQuery = selectStmt
 	}
} else {
 	sqlQuery = selectStmt
}
log.Debugln("\n--\n" + sqlQuery)
return sqlQuery, queryValues
```

Reply:
k, I'll update it